### PR TITLE
bsd: Fix Poll(0) returning ETIMEDOUT instead of SUCCESS

### PR DESCRIPTION
**Pull request by Thog**
This was an oversight of the implementation.

### CODE REVIEW

**Review by riperiperi — APPROVED** [review]
Nice catch.

File: Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs [hunk 1]
```diff
@@ -336,6 +336,12 @@ bool IsUnexpectedLinuxError(LinuxError error)
                 context.Memory.Write(outputBufferPosition + (ulong)(i * Unsafe.SizeOf<PollEventData>()), events[i].Data);
             }
 
+            // In case of non blocking call timeout should not be returned.
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs:339]: It doesn't look like this is checking if the socket is blocking or not.

Comment (Thog) [reply Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs:339]: non blocking in poll case isn't the same as socket blocking

Comment (gdkchan) [reply Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs:339]: I guess a "non blocking call" is a call where the timeout is 0 then?

Comment (Thog) [reply Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs:339]: exactly yeah

**Pull request merged**